app/pb_kits/playbook/pb_distribution_bar/distribution_bar.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+module Playbook
+  module PbDistributionBar
+    class DistributionBar < Playbook::PbKit::Base
+      PROPS = %i[configured_classname
+                 configured_data
+                 configured_id
+                 configured_size
+                 configured_values].freeze
+
+      def initialize(classname: default_configuration,
+                     data: default_configuration,
+                     id: default_configuration,
+                     size: default_configuration,
+                     values: default_configuration)
+        self.configured_classname = classname
+        self.configured_data = data
+        self.configured_id = id
+        self.configured_size = size
+        self.configured_values = values
+      end
+
+      def size
+        size_options = %w[lg sm]
+        one_of_value(configured_size, size_options, "lg")
+      end
+
+      def values
+        default_value(values_to_percents, [1])
+      end
+
+      def to_partial_path
+        "pb_distribution_bar/distribution_bar"
+      end
+
+    private
+
+      DEFAULT = Object.new
+      private_constant :DEFAULT
+      def default_configuration
+        DEFAULT
+      end
+      attr_accessor(*PROPS)
+
+      def normalize_characters(value)
+        return value.gsub!(/(\d\.\d)|[^a-zA-Z\d]/, '\\1').to_i if value.is_a? String
```

Comment:
What are we doing with this regex here? First part matches any digit followed by a period followed by another digit, I am not sure what we are trying to match on the second part, and why are we replacing matches with `'\\1'`?

I would avoid checking the object type if we can, checking it is a smell.

Reply:
We are able to take any value and normalize it:

`values: [3242, "2425.60", 8292, "$5,662.98"]`